iml-gui/crate/src/components/command_modal.rs

```diff
@@ -194,23 +197,49 @@ pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg, GMsg>)
                 schedule_fetch_tree(model, orders);
             }
         }
+        Msg::CancelJob(job_id) => {
+            if let Some(job) = model.jobs.get(&job_id) {
+                if let Some(ct) = find_cancel_transition(job) {
+                    if model.cancelling_jobs.insert(job_id) {
+                        let fut = apply_job_transition(job_id, ct.clone());
+                        orders.skip().perform_cmd(fut);
+                    }
+                }
+            }
+        }
+        Msg::CancelledJob(job_id, job_result) => match *job_result {
+            Ok(_) => {
+                model.cancelling_jobs.remove(&job_id);
+            }
+            Err(e) => {
+                model.cancelling_jobs.remove(&job_id);
```

Comment:
`model.cancelling_jobs.remove(&job_id);` is common for the both branches.

Could be 
```rust
model.cancelling_jobs.remove(&job_id);
if let Err(e) = *job_result {
  error!(...);
  orders.skip();
}
```

Reply:
Ok, if you find this more nice. I find less :-)